src/app/drive/services/worker.service/downloadWorkerHandler.ts

```diff
@@ -140,17 +130,43 @@ export class DownloadWorkerHandler {
 
       case 'success': {
         const { fileId } = messageData;
-        await writer.close();
-        resolve(fileId);
+        const writer = this.writers.get(downloadId);
+
+        if (writer) {
+          await writer.close();
+          this.writers.delete(downloadId);
+          this.fileStreams.delete(downloadId);
+        }
+
         worker.terminate();
+        removeAbortListener();
```

Comment:
To here, the code is the same in success, abort and error cases. I would add a private method called `handleDownloadCleanup` or something like that to avoid code duplication